src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageDirectiveFeature.cs

```diff
@@ -11,18 +11,43 @@ public static class PageDirectiveFeature
     {
         public static bool TryGetRouteTemplate(RazorProjectItem projectItem, out string template)
         {
+            if (projectItem == null)
+            {
+                throw new ArgumentNullException(nameof(projectItem));
+            }
+
             const string PageDirective = "@page";
 
+            var stream = projectItem.Read();
+
+            if (stream == null)
+            {
+                throw new ArgumentOutOfRangeException($"{nameof(projectItem)}.{nameof(projectItem.Read)} can't return null.");
+            }
+
             string content;
-            using (var streamReader = new StreamReader(projectItem.Read()))
+            using (var streamReader = new StreamReader(stream))
             {
-                content = streamReader.ReadToEnd();
+                content = streamReader.ReadToEnd().TrimStart();
             }
 
             if (content.StartsWith(PageDirective, StringComparison.Ordinal))
             {
-                var newLineIndex = content.IndexOf(Environment.NewLine, PageDirective.Length);
-                template = content.Substring(PageDirective.Length, newLineIndex - PageDirective.Length).Trim();
+                var endOfDirective = content.IndexOf(Environment.NewLine, PageDirective.Length);
+                if (endOfDirective < 0)
+                {
+                    var otherNewLine = Environment.NewLine == "/r/n" ? "/n" : "/r/n";
+                    endOfDirective = content.IndexOf(otherNewLine, PageDirective.Length);
+                }
+
+                // No Newlines, read to end of content
+                if (endOfDirective < 0)
+                {
+                    endOfDirective = content.Length;
+                }
+
+                template = content.Substring(PageDirective.Length, endOfDirective - PageDirective.Length).Trim().Trim('"');
```

Comment:
This will trim an arbitrary number of double-quotes from the start and end of the string. That is, the string `""""""""hi, mom"""""""""""""""""""` will become `hi, mom`. Is that intended here?